### PR TITLE
Add view to generate invoice for contribution

### PR DESCRIPTION
**Pull request by zoek1**
##### Description

When a contribution is made to Gitcoin Grant, an invoice is created using the Gitcoin Bounties template.

1. Each contribution has an invoice and is displayed only to contribution owner:
Test Grant _ Gitcoin

2. Display invoice contribution:
Grow Open Source _ Gitcoin (1)

##### Refers/Fixes

#4264

### CODE REVIEW

File: app/grants/templates/grants/invoice.html [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+          {% if total %}
+             {% trans "Total:" %} ${{total|floatformat:2}}
+          {%else%}
+             {% trans "Total:" %} $0.00
+          {%endif%}
+        </td>
+      </tr>
+    </table>
+  </div>
+</body>
```
Comment (thelostone-mc) [inline app/grants/templates/grants/invoice.html]: indent this by +1 ?

File: app/grants/templates/grants/invoice.html [hunk 2]
```diff
@@ -0,0 +1,212 @@
+{% load i18n static %}
+<!doctype html>
+<html>
+<head>
+  <meta charset="utf-8">
+    {% include 'shared/head.html' %}
+    {% include 'shared/cards.html' %}
+
```
Comment (thelostone-mc) [inline app/grants/templates/grants/invoice.html]: ^ deindent these by 1

**Comment by danlipert** [comment]
Hey Miguel - thanks for your PR - next time if you are submitting a PR thats still a WIP, please use the draft PR function - thanks!

**Comment by zoek1** [comment]
Sure, I'll add the screenshot and update the PR description.

**Comment by danlipert** [comment]
@zoek1 is this ready for review now?

**Comment by zoek1** [comment]
Yep, it's ready for review!
@danlipert

File: app/grants/templates/grants/invoice.html [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+          </td>
+        </tr>
+
+        <tr class="details">
+          <td>
+            {% trans "Directly to counterparties via Ethereum Blockchain" %}
+          </td>
+
+          <td>
+            0x2af47a65da8cd66729b4209c22017d6a5c2d2400
```
Comment (danlipert) [inline app/grants/templates/grants/invoice.html]: Whats up with this hardcoded value?

Comment (zoek1) [reply app/grants/templates/grants/invoice.html]: I based this invoice on the invoice template for bounties, I took the value from there. As I understand this address represents the gitcoin address to manage such transactions. This value is hardcoded in other places in the codebase, what is recommended in this cases? @danlipert

Comment (danlipert) [reply app/grants/templates/grants/invoice.html]: So I took a look and actually its the Bounties Network v1 Contract Address, which doesn't make sense here... lets use the grant contract address for each grant instead.

File: app/grants/views.py [hunk 4]
```diff
@@ -842,3 +842,22 @@ def is_verified(tx_details, tx_hash, tx_amount, network):
         )
 
     return get_json_response("Wrong request.", 400)
+
+
+def invoice(request, contribution_pk):
+    contribution = Contribution.objects.prefetch_related('subscription', 'subscription__grant').get(pk=contribution_pk)
+
+    # only allow invoice viewing if admin or if grant contributor
+    is_contributor = request.user.username.lower().lstrip('@') == contribution.subscription.contributor_profile.handle.lower().lstrip('@')
```
Comment (danlipert) [inline app/grants/views.py]: hmm.. probably can do this logic simpler like `has_view_privs = request.user.is_staff or request.user.profile == contribution.subscription.contributor_profile`

Comment (zoek1) [reply app/grants/views.py]: Cool,  i just changed!

**Pull request merged**